packages/app-project/src/screens/ClassifyPage/components/ClassifierWrapper/ClassifierWrapper.js

```diff
@@ -43,6 +46,17 @@ export default function ClassifierWrapper({
     console.error('Classifier error', error)
   }
 
+  function onSubjectChange(subject) {
+    const { query } = router
+    const baseURL = `/${query.owner}/${query.project}/classify`
+    if (query.subjectID && subject.id !== query.subjectID) {
+      const newSubjectRoute = `${baseURL}/workflow/${workflowID}/subject-set/${subjectSetID}/subject/${subject.id}`
+      const href = addQueryParams(newSubjectRoute, router)
+      const as = href
+      router.push(href, as, { shallow: true })
+    }
```

Comment:
This will break URLs like https://frontend.preview.zooniverse.org/projects/brooke/i-fancy-cats/classify/workflow/693/subject/4224?env=staging&demo=true so maybe the check should be explicitly for a workflow that uses indexed subjects.